File: simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/NotificationKey.java
package com.zhuinden.navigationexamplefrag;

import com.google.auto.value.AutoValue;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class NotificationKey extends BaseKey {
    public static NotificationKey create() {
        return new AutoValue_NotificationKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new NotificationFragment();
    }
}

File: simple-stack-example-kotlin/src/main/kotlin/com/zhuinden/simplestackexamplekotlin/NotificationKey.kt
package com.zhuinden.simplestackexamplekotlin

import android.os.Parcelable
import paperparcel.PaperParcel

/**
 * Created by Owner on 2017.11.13.
 */
@PaperParcel
object NotificationKey : BaseKey() {
    override fun createFragment(): BaseFragment = NotificationFragment()

    @JvmField val CREATOR: Parcelable.Creator<NotificationKey> = PaperParcelNotificationKey.CREATOR

    override fun toString(): String = this::class.java.name // you NEED to implement this in an `object` because of Kotlin!
}
